### ISSUE TITLE
Date ranges should be allowed for documents

### ISSUE THREAD

**Issue by martindholmes** [projectEndings/staticSearch#15]
Documents may of course not have precise dates, so it should be possible to specify a date range for a document in the metadata and have the search filters handle the range correctly. We should follow ISO 8601 in using a slash to separate the bounds of the interval:

`2019-01-23/2019-05-30`

and the search filtering should test true (i.e. match the document) when the start of the range specified on the search form comes before the end of the interval, and the end of the specified range comes after the start of the specified interval. So with a document date as specified above, the following search ranges should match the document:

`2019-02/2019-06`
`2019-01-01/2019-04-30`
`2019-02/2019-03`

In other words, any overlap between the document date range and the search date range should constitute a match for the filter.

**Comment by martindholmes** [comment]
I believe this functionality is now working as of commit 6c6b127. Tested in Keats; waiting for more detailed testing in WEA.

**Comment by joeytakeda** [comment]
Works perfectly in WEA, so I think this can be closed.
